backend/ddang/src/main/java/com/ddang/ddang/region/infrastructure/api/RestTemplateInitRegionProcessor.java

```diff
@@ -0,0 +1,131 @@
+import com.ddang.ddang.region.infrastructure.api.exception.RegionApiException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Component
+@RequiredArgsConstructor
+public class RestTemplateInitRegionProcessor implements InitializationRegionProcessor {
+
+    private static final String REGION_API_DOMAIN = "https://sgisapi.kostat.go.kr";
+    private static final String REGION_CODE_NAME = "cd";
+    private static final String ACCESS_TOKEN_NAME = "accessToken";
+    private static final String SERVICE_KEY_NAME = "consumer_key";
+    private static final String SECRET_KEY_NAME = "consumer_secret";
+    private static final String AUTHENTICATION_URL = "/OpenAPI3/auth/authentication.json";
+    private static final String REGION_URL_NAME = "/OpenAPI3/addr/stage.json";
+
+    @Value("${open.api.region.service}")
+    private String serviceSecret;
+
+    @Value("${open.api.region.key}")
+    private String keySecret;
+
+    private final RestTemplate restTemplate;
+
+    @Override
+    public List<Region> requestRegions() {
+        final String accessToken = requestApiAccessToken();
+        final List<ResultApiRegionResponse> firstRegionsResponses = requestApiRegionResponse(
+                null,
+                accessToken
+        );
+
+        final List<Region> totalRegions = new ArrayList<>();
+
+        for (ResultApiRegionResponse firstRegionResponse : firstRegionsResponses) {
+            final Region firstRegion = firstRegionResponse.toEntity();
+            final List<ResultApiRegionResponse> secondRegionsResponses = requestApiRegionResponse(
+                    firstRegionResponse.getRegionCode(),
+                    accessToken
+            );
+
+            for (ResultApiRegionResponse secondRegionsResponse : secondRegionsResponses) {
+                final Region secondRegion = secondRegionsResponse.toEntity();
+                final List<ResultApiRegionResponse> thirdRegionsResponses = requestApiRegionResponse(
+                        secondRegionsResponse.getRegionCode(),
+                        accessToken
+                );
+
+                firstRegion.addSecondRegion(secondRegion);
+
+                for (ResultApiRegionResponse thirdRegionsResponse : thirdRegionsResponses) {
+                    final Region thirdRegion = thirdRegionsResponse.toEntity();
+                    secondRegion.addThirdRegion(thirdRegion);
+                }
+            }
+
+            totalRegions.add(firstRegion);
+        }
+
+        return totalRegions;
+    }
+
+    private String requestApiAccessToken() {
+        URI authenticationUri = UriComponentsBuilder.fromUriString(REGION_API_DOMAIN)
```

Comment:
```suggestion
        final URI authenticationUri = UriComponentsBuilder.fromUriString(REGION_API_DOMAIN)
```

`authenticationUri`에 final이 빠진 듯해요!

Reply:
앗 `final`...추가하겠습니다!